### DISCUSSION TITLE
Compatibility with Vladmandic's Auto1111 fork?

### DISCUSSION THREAD

**Discussion by Eljoseto** [hako-mikan/sd-webui-supermerger#62] [Q&A]
Hi! hako-mikan san, I recently moved from AUTO1111 main repo to [this fork](https://github.com/vladmandic/automatic/tree/master), as the main branch has been stale for a long time. Vlad's fork has all the recent pull request and is being actively mantained.

That said, I'm having problems runing the SurperMerger on that fork, maybe becaus many options have been reorganized. In particular I'm getting this error:

> Traceback (most recent call last):
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
>     output = await app.get_blocks().process_api(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\blocks.py", line 1075, in process_api
>     result = await self.call_function(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\blocks.py", line 884, in call_function
>     prediction = await anyio.to_thread.run_sync(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
>     return await get_asynclib().run_sync_in_worker_thread(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
>     return await future
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
>     result = context.run(func, *args)
>   File "D:\MachineLearning\VladMandic\extensions\sd-webui-supermerger\scripts\mergers\mergers.py", line 73, in smergegen
>     usemodelgen(theta_0,model_a,currentmodel)
>   File "D:\MachineLearning\VladMandic\extensions\sd-webui-supermerger\scripts\mergers\model_util.py", line 798, in usemodelgen
>     devices.unet_needs_upcast = shared.cmd_opts.upcast_sampling and devices.dtype == torch.float16 and devices.dtype_unet == torch.float16
> AttributeError: 'Namespace' object has no attribute 'upcast_sampling'

The start arguments are: `--autolaunch --listen --port 7861 --enable-insecure`

System Info:

torch:  "2.0.0+cu118 autocast half",
gpu:   "NVIDIA GeForce RTX 3060 (12Gb) (compute_37) (8, 6)",
cuda:  "11.8",
cudnn:  8700
memory:  31.95
xformers:  "0.0.17",
accelerate:  "0.18.0",
transformers:  "4.26.1"
arch:  "AMD64",
cpu:  "AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD",
system:  "Windows",
python: 3.10.11,
gradio: 3.23.0,

**Comment by hako-mikan** [comment]
may be fixed

**Reply by Eljoseto** [reply]
Now it works but only the Weight Sum. The rest of the methodes, and the save option give this error:

> > Traceback (most recent call last):
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
>     output = await app.get_blocks().process_api(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\blocks.py", line 1075, in process_api
>     result = await self.call_function(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\gradio\blocks.py", line 884, in call_function
>     prediction = await anyio.to_thread.run_sync(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
>     return await get_asynclib().run_sync_in_worker_thread(
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
>     return await future
>   File "D:\MachineLearning\VladMandic\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
>     result = context.run(func, *args)
>   File "D:\MachineLearning\VladMandic\extensions\sd-webui-supermerger\scripts\mergers\mergers.py", line 65, in smergegen
>     result,currentmodel,modelid,theta_0,metadata = smerge(
>   File "D:\MachineLearning\VladMandic\extensions\sd-webui-supermerger\scripts\mergers\mergers.py", line 174, in smerge
>     theta_2 = load_model_weights_m(model_c,False,False,save).copy()
>   File "D:\MachineLearning\VladMandic\extensions\sd-webui-supermerger\scripts\mergers\mergers.py", line 492, in load_model_weights_m
>     return sd_models.read_state_dict(checkpoint_info.filename,"cuda")
> TypeError: read_state_dict() takes 1 positional argument but 2 were given

**Comment by hako-mikan** [comment]
I see that someone reported a similar error, and I couldn't figure out the cause of the error, but he was using the forked version.
May be fixed.

**Reply by Eljoseto** [reply]
Thanks You so much! Your last update is working now
